### PR TITLE
fix: test updates for az defaults

### PR DESCRIPTION
**Pull request by tsc-buddy**
## Description

Addressing the AZ defaults now that the PSRule test bug has been addressed [here](https://github.com/microsoft/PSRule/issues/1900#issuecomment-2356575578).

## Pipeline Reference

| Pipeline |
| -------- |
|[avm.res.web.serverfarm](https://github.com/tsc-buddy/bicep-registry-modules/actions/workflows/avm.res.web.serverfarm.yml)|

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [x] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Review by pankajagrawal16 — DISMISSED** [review]

File: avm/res/web/serverfarm/main.bicep [hunk 1]
```diff
@@ -16,10 +16,10 @@ param name string
   'I1v2'
   '''
 })
-param skuName string
+param skuName string = 'P1v3'
```
Comment (eriqua) [inline avm/res/web/serverfarm/main.bicep:19]: Hey @tsc-buddy thanks a lot for this update and also for opening and following up on a PSRule fix.

Is it intentional to set this specific value as default? If so, please also:
- change the parameter type to Optional in the description
- remove the parameter from defaults test, as they should only use required parameters (optional parameters are meant to use their defaults in this case)

File: avm/res/web/serverfarm/main.bicep [hunk 2]
```diff
@@ -16,10 +16,10 @@ param name string
...
 
 @description('Required. Number of workers associated with the App Service Plan.')
-param skuCapacity int
+param skuCapacity int = 3
```
Comment (eriqua) [inline avm/res/web/serverfarm/main.bicep:22]: Same comment of skuName

File: avm/res/web/serverfarm/main.bicep (hunk 1 above)
Comment (tsc-buddy) [reply avm/res/web/serverfarm/main.bicep:19]: Yes it was intentional, simply to make a default deployment leverage Availability zones.

I can update the test.

Comment (eriqua) [reply avm/res/web/serverfarm/main.bicep:19]: All good for me if that is the intention @tsc-buddy Just double checking.
I'd still suggest the 2 points above if a default value is set, i.e. make it Optional and remove from defaults test.

**Comment by tsc-buddy** [comment]
Roger that! Thanks for the recommendation. Will make that change now. :) - Once merged I will watch the Action to make sure it builds on the AVM subscription OK and publishes as last time there was a quota issue.

File: avm/res/web/serverfarm/main.bicep [hunk 3]
```diff
@@ -18,7 +18,7 @@ param name string
 })
 param skuName string = 'P1v3'
 
-@description('Required. Number of workers associated with the App Service Plan.')
+@description('Optional. Number of workers associated with the App Service Plan. This defaults to 3, to leverage availability zones..')
```
Comment (eriqua) [inline avm/res/web/serverfarm/main.bicep:21]: Minor typo, please bare with me :) 
```suggestion
@description('Optional. Number of workers associated with the App Service Plan. This defaults to 3, to leverage availability zones.')
```

Comment (tsc-buddy) [reply avm/res/web/serverfarm/main.bicep]: All done. Thanks @eriqua, love the eagle eyes!

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**